src/Lykke.AlgoStore.MatchingEngineAdapter.Services/Listening/ConsumerLoadBalancer.cs

```diff
@@ -0,0 +1,110 @@
+﻿using Lykke.AlgoStore.MatchingEngineAdapter.Core.Services.Listening;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Common.Log;
+using JetBrains.Annotations;
+using Lykke.AlgoStore.MatchingEngineAdapter.Core.Services;
+
+namespace Lykke.AlgoStore.MatchingEngineAdapter.Services.Listening
+{
+    /// <summary>
+    /// Manages a set of <see cref="ConsumingWorker"/> by adding or removing instances based on the
+    /// current message load
+    /// </summary>
+    internal class ConsumerLoadBalancer : IDisposable
```

Comment:
I see no reason to use custom balancers here. Why not just to use async await and rely on scheduler?